common/src/main/java/net/dodogang/plume/cosmetic/CosmeticPlayerData.java

```diff
@@ -0,0 +1,35 @@
+package net.dodogang.plume.cosmetic;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+public class CosmeticPlayerData {
+    private final @NotNull UUID playerUUID;
+    private final @NotNull Map<CosmeticSlot, Cosmetic> cosmeticToSlot;
+
+    private CosmeticPlayerData(UUID playerUUID, Map<CosmeticSlot, Cosmetic> cosmeticToSlot) {
+        this.playerUUID = playerUUID;
+        this.cosmeticToSlot = cosmeticToSlot;
+    }
+
+    public static CosmeticPlayerData of(UUID playerUUID, Cosmetic... cosmetics) {
```

Comment:
Since the private final fields are labeled `@NotNull`, the constructor and static 'constructor' fields should also be labeled `@NotNull`.